spec/integ/pm.spec.js

```diff
@@ -299,6 +299,49 @@ describe("IRC-to-Matrix PMing", function() {
         yield createRoomPromise;
         yield sentMessagePromise;
     }));
+
+    it("should not create multiple matrix rooms when several PMs are received in quick succession",
+    test.coroutine(function*() {
+        let count = 0;
+        // mock create room impl
+        let createRoomPromise = new Promise(function(resolve, reject) {
+            sdk.createRoom.andCallFake(function(opts) {
+                count++;
+                expect(count).toEqual(1);
+                resolve();
+                return Promise.resolve({
+                    room_id: tCreatedRoomId
+                });
+            });
+        });
+
+        // mock send message impl
+        let sentMessagePromise = new Promise(function(resolve, reject) {
+            sdk.sendEvent.andCallFake(function(roomId, type, content) {
+                expect(roomId).toEqual(tCreatedRoomId);
+                expect(type).toEqual("m.room.message");
+                expect(content).toEqual({
+                    body: tText,
+                    msgtype: "m.text"
+                });
+                resolve();
+                return Promise.resolve({});
+            });
+        });
+
+        // find the *VIRTUAL CLIENT* (not the bot) and send the irc message
+        let client = yield env.ircMock._findClientAsync(
+            roomMapping.server, tRealMatrixUserNick
+        );
+
+        // Send several messages, almost at once, to simulate a race
+        for (var i = 0; i < 10; i++) {
+            client.emit("message", tRealIrcUserNick, tRealMatrixUserNick, tText);
+        }
+
+        yield createRoomPromise;
+        yield sentMessagePromise;
```

Comment:
We expect this to send **10** messages but we're only waiting for one. Why? This test can race in its current form because:
 - Send 10 messages.
 - #1 arrives
 - #2 arrives
 - #1 creates a room.
 - #1 sends the message.
 - PROMISES RESOLVED : TEST PASS :+1: 
 - #2 creates a room. `expect()` catches this but test has already passed so NOPs.
 - #2 sends the message.